pallets/staking/src/lib.rs

```diff
@@ -1997,33 +1960,46 @@ decl_module! {
             let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
             let stash = &ledger.stash;
             ensure!(!targets.is_empty(), Error::<T>::EmptyTargets);
+			ensure!(targets.len() <= MAX_NOMINATIONS, Error::<T>::TooManyTargets);
+
+            let old = Nominators::<T>::get(stash).map_or_else(Vec::new, |x| x.targets);
+
             // A Claim_key can have multiple claim value provided by different claim issuers.
             // So here we iterate every CDD claim provided to the nominator If any claim is greater than
             // the threshold value of timestamp i.e current_timestamp + Bonding duration
             // then nominator is added into the nominator pool.
 
-            if let Some(nominate_identity) = <Identity<T>>::get_identity(stash) {
-                let leeway = Self::get_bonding_duration_period() as u32;
-                if <Identity<T>>::fetch_cdd(nominate_identity, leeway.into()).is_some() {
-                    let targets = targets.into_iter()
-                        .take(MAX_NOMINATIONS)
-                        .map(T::Lookup::lookup)
-                        .collect::<result::Result<Vec<T::AccountId>, _>>()?;
-
-                    let nominations = Nominations {
-                        targets: targets.clone(),
-                        // initial nominations are considered submitted at era 0. See `Nominations` doc
-                        submitted_in: Self::current_era().unwrap_or(0),
-                        suppressed: false,
-                    };
-
-                    // Polymesh-note: Decrement the running count by 1.
-                    Self::release_running_validator(&stash);
-                    <Validators<T>>::remove(stash);
-                    <Nominators<T>>::insert(stash, &nominations);
-                    Self::deposit_event(RawEvent::Nominated(nominate_identity, stash.clone(), targets));
-                }
-            }
+            let nominate_identity = match <Identity<T>>::get_identity(stash) {
+                Some(ni)
+                    if <Identity<T>>::fetch_cdd(ni, (Self::get_bonding_duration_period() as u32).into()).is_some()
+                    => ni,
+                _ => return Ok(()),
+            };
+            let targets = targets.into_iter()
+                .map(|t| T::Lookup::lookup(t).map_err(DispatchError::from))
+                .map(|n| n.and_then(|n| if old.contains(&n) || !Validators::<T>::get(&n).blocked {
+                    Ok(n)
+                } else {
+                    Err(Error::<T>::BadTarget.into())
+                }))
```

Comment:
Some new logic here.

Reply:
Except for the refactoring of our CDD logic, this is taken from substrate, so I'd like to avoid changing their code, but this is what the code does:
1. Limits the number of nominators.
2. Ensures that *new* nominators are not accepted if the validator has blocked this.